File: Medium/Painting the Fence/painting-the-fence.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
#include<stdio.h>
#include<math.h>
using namespace std;

// } Driver Code Ends
class Solution{
    public:
    long long countWays(int n, int k){
        // code here
        long long ans=1;
        long long prev=1;
        long long mod=1000000007;
        for(int i=1;i<n;i++){
            long long temp=(k-1)*ans;
            ans=(prev+temp)%mod;
            prev=temp;
        }
        return (ans*k)%mod;
    }
};

//{ Driver Code Starts.

int main()
{
	
	int t;
	cin>>t;
	while(t--)
	{
		int n,k;
		cin>>n>>k;
		Solution ob;
		cout<<ob.countWays(n,k)<<endl;
	}
	return 0;
}
// } Driver Code Ends

File: Medium/Painting the Fence/painting-the-fence.py
#User function Template for python3

class Solution:
    def countWays(self,n,k):
        #code here.
        mod=10**9 + 7
        diff=k
        total=k
        for i in range(2,n+1) :
            same=diff
            diff=((k-1)*total)%mod
            total=(same+diff)%mod
        return total



#{ 
 # Driver Code Starts

#Initial Template for Python 3




t=int(input())
for _ in range(0,t):
    x=list(map(int,input().split()))
    n=x[0]
    k=x[1]
    ob = Solution()
    ans=ob.countWays(n,k)
    print(ans)

# } Driver Code Ends